lib/hl7/datetime_components.rb

```diff
@@ -2,26 +2,29 @@
 
 module HL7
   class DatetimeComponents
-    attr_reader :year, :month, :day, :hour, :minute, :second, :offset_seconds
+    attr_reader :year, :month, :day, :hour, :minute, :second, :fraction, :offset_seconds
 
-    def initialize(year:, month: 1, day: 1, hour: 0, minute: 0, second: 0, offset_seconds: 0)
+    def initialize(year, month = nil, day = nil, hour = nil, minute = nil, second = nil, fraction = nil, offset_seconds: nil)
       @year = year
-      @month = (month || 1)
-      @day = (day || 1)
-      @hour = (hour || 0)
-      @minute = (minute || 0)
-      @second = (second || 0)
-      @offset_seconds = (offset_seconds || 0)
+      @month = month
+      @day = day
+      @hour = hour
+      @minute = minute
+      @second = second
+      @fraction = fraction
+      @offset_seconds = offset_seconds
       validate!
     end
 
     def to_time(zone:)
+      full_seconds = second.to_i + fraction.to_f
+
       if offset_seconds.nil?
         Time.use_zone(zone) do
```

Comment:
is there a significant difference between how these two case are handled? i.e.e why couldn't you use `#in_time_zone` for both?

Reply:
Doesn't work. Problem is that `Time.zone.local` doesn't accept a UTC offset. What it _should_ do is use the offset to generate the point-in-time and then convert to the desired zone, but it just doesn't.

Reply:
ahh I see, thanks 👍